core/src/main/java/org/apache/iceberg/PartitionStats.java

```diff
@@ -78,7 +78,16 @@ public int equalityDeleteFileCount() {
     return equalityDeleteFileCount;
   }
 
+  /**
+   * @deprecated since 1.9.0, will be removed in 1.10.0, use {@link #totalRecordCountOptional()}
+   *     instead.
+   */
+  @Deprecated
   public long totalRecordCount() {
+    return totalRecordCount == null ? 0 : totalRecordCount;
+  }
+
+  public Long totalRecordCountOptional() {
```

Comment:
maybe `totalRecords()`?

Reply:
I like it. Updated. 

Also, after rebased and observed test failure due to partition stats writer code. Handled them also.